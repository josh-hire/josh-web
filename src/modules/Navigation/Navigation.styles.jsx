import { createStyles } from '@mantine/core';

export const useStyles = createStyles((theme) => ({
  navbar: {
    backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[6] : theme.white,
    paddingBottom: 0,
  },

  header: {
    padding: theme.spacing.md,
    paddingTop: 0,
    marginLeft: -theme.spacing.md,
    marginRight: -theme.spacing.md,
    color: theme.colorScheme === 'dark' ? theme.white : theme.black,
  },

  links: {
    marginLeft: -theme.spacing.md,
    marginRight: -theme.spacing.md,
  },

  footer: {
    marginLeft: -theme.spacing.md,
    marginRight: -theme.spacing.md,
    borderTop: `1px solid ${
      theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]
    }`,
  },
}));
