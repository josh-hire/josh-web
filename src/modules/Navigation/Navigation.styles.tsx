import { createStyles } from '@mantine/core';

export const useStyles = createStyles((theme) => ({
  navbar: {
    backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[6] : theme.white,
    paddingBottom: 0,
  },

  header: {
    padding: theme.spacing.md,
    paddingTop: 0,
    marginLeft: -theme.spacing.md,
    marginRight: -theme.spacing.md,
    color: theme.colorScheme === 'dark' ? theme.white : theme.black,
  },

  links: {
    marginLeft: -theme.spacing.md,
    marginRight: -theme.spacing.md,
  },

  footer: {
    marginLeft: -theme.spacing.md,
    marginRight: -theme.spacing.md,
  },
}));
