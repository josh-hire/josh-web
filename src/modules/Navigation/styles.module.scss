.navbar {
  background-image: url('/assets/bg/bg-navigation.svg');
  background-repeat: no-repeat;
  background-position: bottom;
  background-size: contain;
  width: var(--width);
}

.hidden {
  width: var(--width);
  padding: 0;
}
