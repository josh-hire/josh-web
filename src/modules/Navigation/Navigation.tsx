import { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Box, clsx, Navbar, ScrollArea } from '@mantine/core';
import { useStyles } from './Navigation.styles';
import styles from './styles.module.scss';

export default function Navigation({ hidden, links,  ...rest }) {
  const { classes } = useStyles();
  const width = useMemo(() => (hidden ? '100px' : '260px'), [hidden]);

  return (
    <Navbar
      className={clsx([styles.navbar, hidden && styles.hidden])}
      p="md"
      style={{ 'width': width }}
      width={{ base: !hidden ? 260 : 100 }}
      {...rest}>
      <Navbar.Section
        className={classes.links}
        component={ScrollArea}
        grow
        sx={{ width: hidden ? '115px' : 'auto' }}>
        <Box py={hidden ? '20px' : '0'}>{links}</Box>
      </Navbar.Section>
    </Navbar>
  );
}

Navigation.propTypes = {
  hidden: PropTypes.bool,
  links:PropTypes.string,
  list: PropTypes.array,
};

Navigation.defaultProps = {
  hidden: false,
  links:'',
  list: [],
};
