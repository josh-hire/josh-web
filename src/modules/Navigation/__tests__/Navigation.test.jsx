import ShallowRenderer from 'react-test-renderer/shallow';
import { default as Navigation } from '../Navigation';

describe('src/components/Navigation', ()=>{
  test('Navigation',()=>{
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Navigation hidden="false" links="https://test.com"/>);
    expect(tree).toMatchSnapshot();
  });
});
