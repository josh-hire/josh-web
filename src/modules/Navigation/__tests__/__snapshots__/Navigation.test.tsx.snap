// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Navigation Navigation 1`] = `
<@mantine/core/Navbar
  className="navbar hidden"
  list={[]}
  p="md"
  style={
    {
      "width": "100px",
    }
  }
  width={
    {
      "base": 100,
    }
  }
>
  <@mantine/core/Section
    className="mantine-1do6r92"
    component={
      {
        "$$typeof": Symbol(react.forward_ref),
        "Autosize": {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "render": [Function],
      }
    }
    grow={true}
    sx={
      {
        "width": "115px",
      }
    }
  >
    <@mantine/core/Box
      py="20px"
    >
      https://test.com
    </@mantine/core/Box>
  </@mantine/core/Section>
</@mantine/core/Navbar>
`;
